### PR TITLE
burger2022reproducibility

### PR DESCRIPTION
**Pull request by kayatb**
## Project Name
burger2022reproducibility
[RE] Reproducibility Study - SCOUTER: Slot Attention-based Classifier for Explainable Image Recognition

## Compiled PDF

Add the link to the compiled pdf here. Example : [Compiled Pdf](2021/burger2022reproducibility/journal/article.pdf)

## Metadata Checklist

- [x] `title` field exists and it starts with `[Re] `.
- [x] `authors` contain author information with affiliations.
- [x] [ORCID id](https://orcid.org/) is provided for authors.
- [x] One contact author is added using `*`.
- [x] Github `url` is provided in `code` section.
- [x] [SWH identifier](https://archive.softwareheritage.org/save/) is provided in `swh`.
- [x] `replication` section is populated with the necessary information about the paper you reproduced.
- [x] `abstract` field is populated with your paper abstract (including double quotes)
- [x] `domain` should contain "ML Reproducibility Challenge 2021"
- [x] `review` section contains the url to the OpenReview forum of your paper.
- [x] Fields `contributors`, `dates`, `article` and `journal` are kept blank, as those will be populated by the Area Chairs / ReScience editors.

## Build Errors
We encountered issues with citations, images and referring to tables. The images aren't displayed at all and referring to tables does not work. In the build.log are various messages of "undefined citations". Furthermore, the header partly overlaps in the pdf.
See [build.log](https://pastebin.com/1znce6Au)
Our ./build_all.sh output can be found [here](https://pastebin.com/THik0erB)

### CODE REVIEW

File: 2021/burger2022reproducibility/openreview/packages.tex [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+\usepackage[utf8]{inputenc} % allow utf-8 input
+%\usepackage[T1]{fontenc}    % use 8-bit T1 fonts
+\usepackage{hyperref}       % hyperlinks
+\usepackage{url}            % simple URL typesetting
+\usepackage{booktabs}       % professional-quality tables
+\usepackage{amsfonts}       % blackboard math symbols
+\usepackage{nicefrac}       % compact symbols for 1/2, etc.
+\usepackage{microtype}      % microtypography
+
+\usepackage{subcaption}
```
Comment (koustuvsinha) [inline 2021/burger2022reproducibility/openreview/packages.tex:19]: From the build logs, the first error is:

> ! Package subcaption Error: This package can't be used in cooperation
> (subcaption)                with the subfig package.

I would recommend removing the subcaption package.

Comment (kayatb) [reply 2021/burger2022reproducibility/openreview/packages.tex]: Hi, thanks for the pointer! After removing the package, the figures and tables are now displayed correctly. Unfortunately, we still face the issue of undefined citations and references and the header overlapping. Do you have any clue what could cause this?

File: 2021/burger2022reproducibility/openreview/content.tex [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+\begin{table}[!h]
+\centering
+\begin{tabular}{lc|lc}
+\hline
+\multicolumn{1}{c}{\textbf{Hyperparameter}} & \textbf{Value}           & \textbf{Hyperparameter} & \multicolumn{1}{l}{\textbf{Value}} \\ \hline
+Epochs                                      & 20                       & Lambda Value            & \{1, 3, 10\}       \\
+Batch Size                                  & 70                       & Slots per Class         & 1                                  \\
+Number of Classes                           & min(N_{classes}, 100) & Power of Slot Loss      & 2                                  \\
```
Comment (koustuvsinha) [inline 2021/burger2022reproducibility/openreview/content.tex:128]: replace `N_{classes}` with `$N_{classes}$`

File: 2021/burger2022reproducibility/openreview/content.tex [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+\begin{table}[!h]
+\centering
+\resizebox{\columnwidth}{!}{
+\begin{tabular}{l|cc|cc|cc|cc|cc|cc}
+\hline
+           & \multicolumn{2}{c|}{\textbf{Area Size}} & \multicolumn{2}{c|}{\textbf{Precision}} & \multicolumn{2}{c|}{\textbf{IAUC}} & \multicolumn{2}{c|}{\textbf{DAUC}} & \multicolumn{2}{c|}{\textbf{Infidelity}} & \multicolumn{2}{c}{\textbf{Sensitivity}} \\
+\textbf{Model}      & original      & reproduced     & original      & reproduced     & original   & reproduced   & original   & reproduced   & original      & reproduced      & original       & reproduced      \\ \hline
+Scouter_+ (\lambda = 1)  & 0.1561        & 0.1564           & 0.8493        & \textcolor{orange}{0.7898}           & 0.7512     & \textcolor{orange}{0.3377}         & 0.1753     & \textcolor{orange}{0.4013}         & 0.0799        & \textcolor{orange}{0.0006}            & 0.0796         & \textcolor{orange}{1.9167}            \\
```
Comment (koustuvsinha) [inline 2021/burger2022reproducibility/openreview/content.tex:201]: In all instances of `Scounter_+`, replace them with `$Scouter_+$`

File: 2021/burger2022reproducibility/openreview/content.tex [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+\begin{table}[!h]
+\centering
+\resizebox{0.55\columnwidth}{!}{
+\begin{tabular}{c|cc|cc|cc}
+\hline
+\textbf{}      & \multicolumn{2}{c|}{\textbf{ImageNet}} & \multicolumn{2}{c|}{\textbf{Con-text}} & \multicolumn{2}{c}{\textbf{CUB-200-2011}} \\
+\textbf{Model} & original          & reproduced         & original          & reproduced         & original           & reproduced           \\ \hline
+FC             & 0.8080            & 0.8086               & 0.6732            & 0.6831 (\pm 0.0156)             & 0.7538             & 0.7824 (\pm 0.0274)                 \\
+Scouter_+       & 0.7991            & 0.7717             & 0.6870            & \textcolor{orange}{0.5492} (\pm 0.0182)             & 0.7362             & \textcolor{orange}{0.4718} (\pm 0.0212)               \\
```
Comment (koustuvsinha) [inline 2021/burger2022reproducibility/openreview/content.tex:224]: In all instances of `\pm`, replace it with `$\pm$`

File: 2021/burger2022reproducibility/journal/bibliography.bib [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    title = {WordNet: An Electronic Lexical Database},
+    author = {Christiane Fellbaum},
+    year = {1998},
+    publisher = {Bradford Books},
+  }
+ 
+@techreport{cub-200,
+	Title = {{The Caltech-UCSD Birds-200-2011 Dataset}},
+	Author = {Wah, C. and Branson, S. and Welinder, P. and Perona, P. and Belongie, S.},
+	Year = {2011}
```
Comment (koustuvsinha) [inline 2021/burger2022reproducibility/journal/bibliography.bib:27]: Missing comma after this entry, which is failing the build.

**Comment by koustuvsinha** [comment]
Hi @kayatb , I have added several comments to fix the build issues. Hopefully it will resolve the issue!

**Comment by kayatb** [comment]
Hi, everything is working now. Thank you kindly for your help!

**Pull request merged**